src/components/popover/tab/keymap/keymap-tab.component.ts

```diff
@@ -1,73 +1,74 @@
-import {Component, Input, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
 
-import {Select2OptionData} from 'ng2-select2/ng2-select2';
+import { Select2OptionData } from 'ng2-select2/ng2-select2';
 
-import {KeyAction, SwitchKeymapAction} from '../../../../config-serializer/config-items/key-action';
-import {Keymap} from '../../../../config-serializer/config-items/Keymap';
-import {Tab} from '../tab';
-
-import {UhkConfigurationService} from '../../../../services/uhk-configuration.service';
+import { KeyAction, SwitchKeymapAction } from '../../../../config-serializer/config-items/key-action';
+import { Keymap } from '../../../../config-serializer/config-items/Keymap';
+import { Tab } from '../tab';
 
 @Component({
     selector: 'keymap-tab',
     template: require('./keymap-tab.component.html'),
-    styles: [require('./keymap-tab.component.scss')]
+    styles: [require('./keymap-tab.component.scss')],
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class KeymapTabComponent implements OnInit, Tab {
     @Input() defaultKeyAction: KeyAction;
+    @Input() keymaps: any;
 
-    private keymaps: Keymap[];
     private keymapOptions: Array<Select2OptionData>;
-    private selectedKeymapIndex: number;
+    private selectedKeymap: Keymap;
 
-    constructor(private uhkConfigurationService: UhkConfigurationService) {
-        this.keymaps = [];
+    constructor() {
         this.keymapOptions = [];
-        this.selectedKeymapIndex = -1;
+        this.selectedKeymap = undefined;
     }
 
     ngOnInit() {
-        this.keymaps = this.uhkConfigurationService.getUhkConfiguration().keymaps.elements;
-
         this.keymapOptions.push({
             id: '-1',
             text: 'Switch to keymap'
         });
 
-        this.keymapOptions = this.keymapOptions.concat(this.keymaps.map(function (keymap: Keymap): Select2OptionData {
+        this.keymapOptions = this.keymaps.map((keymap: Keymap): Select2OptionData => {
             return {
-                id: keymap.id.toString(),
+                id: keymap.abbreviation,
                 text: keymap.name
             };
-        }));
+        });
 
         this.fromKeyAction(this.defaultKeyAction);
     }
 
     // TODO: change to the correct type when the wrapper has added it.
     onChange(event: any) {
-        this.selectedKeymapIndex = +event.value;
+        if (event.value === '-1') {
+            this.selectedKeymap = undefined;
+        } else {
+            this.selectedKeymap = this.keymaps.find((keymap: Keymap) => keymap.abbreviation === event.value);
+        }
     }
 
     keyActionValid(): boolean {
-        return this.selectedKeymapIndex >= 0;
+        return this.selectedKeymap !== undefined;
```

Comment:
` return !!this.selectedKeymap;` to work correctly when `selectedKeymap` is either `null` or `undefined`.